### PR TITLE
Include symlinks in Finder

### PR DESCRIPTION
**Pull request by iluuu1994**
| Q             | A
| ------------- | ---
| Fixed tickets | #40

According to [this comment](https://github.com/lavary/crunz/issues/40#issuecomment-631612899) a workaround for Crunz not supporting multiple tasks directories is using symlinks. Unfortunately, this is currently not working because the `RecursiveDirectoryIterator` requires the `FOLLOW_SYMLINKS` to actually look through the contents of the symlink.

Does this change require a test? We'd have to extend `Filesystem` with a method to create a symlink just for the test.

### CODE REVIEW

**Comment by PabloKowalczyk** [comment]
Hello @iluuu1994, thanks for your PR.

> Does this change require a test?

Yes, please.

File: tests/Unit/Finder/FinderTest.php [hunk 1]
```diff
@@ -15,6 +15,10 @@ final class FinderTest extends TestCase
     private $filesystem;
     /** @var Path */
     private $tasksDirectory;
+    /** @var Path */
+    private $symlinkDirectory;
```
Comment (iluuu1994) [inline tests/Unit/Finder/FinderTest.php:19]: Target/source might be clearer.

File: src/Filesystem/Filesystem.php [hunk 2]
```diff
@@ -148,4 +148,26 @@ public function readContent($filePath)
 
         return $content;
     }
+
+    /**
+     * Creates a symbolic link or copy a directory.
+     */
+    public function symlink(string $originDir, string $targetDir): void
```
Comment (iluuu1994) [inline src/Filesystem/Filesystem.php:155]: Should we mark this as internal?

**Comment by PabloKowalczyk** [comment]
Finally got some time to switch TravisCI to GitHub Actions, so please to rebase first.

File: tests/Unit/Finder/FinderTest.php [hunk 3]
```diff
@@ -91,4 +112,25 @@ private function createFiles(Path ...$files): void
                 ->dumpFile($path->toString(), $content);
         }
     }
+
+    /**
+     * @test
+     */
+    public function findFilesInSymlinkedFolder(): void
```
Comment (PabloKowalczyk) [inline tests/Unit/Finder/FinderTest.php:119]: Actually you could prepare directory structure (for example in "tests/resources/fixtures/") in `tests` directory, with this change code will be simpler and there is no need for `Filesystem::symlink` method.

Comment (iluuu1994) [reply tests/Unit/Finder/FinderTest.php:109]: Sorry for the delay. Did just that.

**Comment by iluuu1994** [comment]
@PabloKowalczyk Is it fine to skip this test for Windows? I don't think committing the symlink as suggested will ever work there.

**Comment by PabloKowalczyk** [comment]
> Is it fine to skip this test for Windows?

According to this thread https://stackoverflow.com/questions/5917249/git-symlinks-in-windows symlinks in Windows can work but this needs extra steps from user so it is fine skip this test in Windows now. Windows tests can be added in the future when needed, we can start without it now.

**Review by PabloKowalczyk — APPROVED** [review]

**Comment by PabloKowalczyk** [comment]
Thanks @iluuu1994.

**Pull request merged**